src/vignette/http/route_helpers.clj

```diff
@@ -35,10 +38,31 @@
                             #"^\/(.*)"
                             "$1")))
 
+(defn route->requested-format
+  "Extract information about the format being explicitely specified in the url"
+  [request-map request]
+  (assoc request-map :requested-format (:format (extract-query-opts request))))
+
+(defn browser-supports-webp? [request]
+  (if-let [vary-string (get-in request [:headers webp-accept-header-name])]
+    (.contains vary-string webp-accept-header-value)))
+
+(defn add-webp-format-option-if-supported
+  [request options]
+  (if (browser-supports-webp? request)
+    (merge options {:format "webp"})
+    options))
+
+(defn autodetect-request-format
+  [request options]
+  (if (empty? (:format options))
+    (add-webp-format-option-if-supported request options)
+    options))
+
 (defn route->options
   "Extracts the query options and moves them to 'request-map'"
   [request-map request]
-  (assoc request-map :options (extract-query-opts request)))
+  (assoc request-map :options (autodetect-request-format request (extract-query-opts request))))
```

Comment:
You can leave this function as-is and use the extracted `:format` value in `route->requested-format`